variables.tf

```diff
@@ -184,40 +179,43 @@ variable "kyverno" {
   default = {}
 }
 
-variable "alloy" {
+variable "grafana_alloy" {
   description = "grafana alloy configuration"
   type = object({
-    enabled = optional(bool, true)
-    loki = optional(
-      object({
-        url      = string
-        username = optional(string, null)
-        password = optional(string, null)
-      }),
-      {
-        url      = null
-        username = null
-        password = null
-      }
-    )
-    loki_scrape_global             = optional(bool, true)
-    loki_collect_self_logs_enabled = optional(bool, false)
-    prometheus = optional(
-      object({
-        url      = string
-        username = optional(string, null)
-        password = optional(string, null)
-      }),
-      {
-        url      = null
-        username = null
-        password = null
-      }
-    )
-    tenant_id                  = optional(string, "default")
-    node_template_file_path    = optional(string, null)
-    cluster_template_file_path = optional(string, null)
+    image = optional(object({
+      repository = optional(string, "grafana/alloy")
+    }), {})
+    metrics = optional(object({
+      endpoint    = optional(string, null)
+      tenant      = optional(string, null)
+      ssl_enabled = optional(bool, false)
+    }), {})
+    cluster = optional(object({
+      enabled  = optional(bool, true)
+      replicas = optional(number, 3)
+      requests = optional(object({
+        cpu    = optional(string, "100m")
+        memory = optional(string, "256Mi")
+      }), {})
+      limits = optional(object({
+        cpu    = optional(string, "100m")
+        memory = optional(string, "256Mi")
+      }), {})
+
+    }), {})
+    node = optional(object({
+      enabled = optional(bool, true)
+      requests = optional(object({
+        cpu    = optional(string, "100m")
+        memory = optional(string, "128Mi")
+      }), {})
+      limits = optional(object({
+        cpu    = optional(string, "100m")
+        memory = optional(string, "256Mi")
+      }), {})
+    }), {})
   })
+  default = {}
```

Comment:
Isn't at least `node.enabled` and  `cluster.enabled` needed in the default to prevent failure in the module's count?